SAC-3349 support TYPO3 v11

DanielSiepmann opened the pull request.

dhuf committed:
Update composer.json

dhuf committed:
v3.0.0 + php7.4

DanielSiepmann committed:
[TASK] Update composer.json and .gitignore

* Support latest TYPO3 composer installed (extra section).
* Remove empty sections.
* Allow installation with TYPO3 11.5.
* Allow installation with expected and tested PHP versions.
* Update .gitignore to match state after composer install.

DanielSiepmann committed:
Initialy setup phpstan

DanielSiepmann committed:
Instal and configure rector

DanielSiepmann committed:
Execute rector

DanielSiepmann committed:
Configure rector to process the whole extension

No need to call `process .`.

DanielSiepmann commented:
I marked the extension to be compatible with our use case.
I executed rector to auto migrate most parts.
I manually checked the small code part.

I did not test the update yet. There are 0 tests within the extension. I also won't do that. Instead I'll add functional tests to the extension to ensure that it works as expected.
I also would add GitHub Actions.

DanielSiepmann committed:
TASK: Add functional tests

In order to ensure functionality is working within TYPO3.
This also eases updates in future as one can execute tests and knows
whether everything is working as expected.
Also removes the need to install within an example or real world
installation for manual testing.

DanielSiepmann committed:
Add CGL

DanielSiepmann committed:
Add GitHub Actions

Fix none valid XML (fix namespace and version).

DanielSiepmann commented:
Done from my sight.

Hannes mentioned there is an v11 version on https://github.com/visol/ext-variables.
I try to rebase and contribute our changes back.

They didn't drop older TYPO3 and PHP, need to extend the CI to also test those versions.

DanielSiepmann commented:
I've created a new branch based on this one, rebased on v11 from the original repo.
The PR is available at: https://github.com/visol/ext-variables/pull/21

I hope they merge it and release a version.
I would keep our fork and use the branch from https://github.com/visol/ext-variables/pull/21 otherwise.

s12r commented on ext_emconf.php:
```diff
@@ -8,35 +8,20 @@
-            'typo3' => '9.5.0-9.5.99',
-            'php' => '7.0.0-7.2.99',
+            'typo3' => '9.5.0-10.9.99',
+            'php' => '7.0.0-7.4.99',
```
Doesn't match with `composer.json`'s
```
  ...
  "typo3/cms-core": "^11.5",
  "php": "8.1.*"
  ...
```
... or have I missed something?

s12r reviewed (changes requested):
found only a version mismatch between `composer.json` and `ext_emconf.php`, the rest seems ok  so far

DanielSiepmann committed:
Fix outdated ext_emconf.php version dependencies

Do not add php and do not add specific typo3 dependency.
This is all resolved by composer and composer installer nowadays.
